plugins/synced_folders/smb/synced_folder.rb

```diff
@@ -168,6 +174,10 @@ def cleanup(machine, opts)
         if machine.env.host.capability?(:smb_cleanup)
           machine.env.host.capability(:smb_cleanup, machine, opts)
         end
+        # Try to unmount all smb mounts, ignore if unmount fails for any reason
+        machine.communicate.sudo("mount -v | grep 'type cifs' | awk '{print $3}' | xargs umount || true")
+        # Clean up any credentials files that might exist
+        machine.communicate.sudo("rm /etc/smb_creds_vgt-*")
```

Comment:
I'm not entirely sure if this is an appropriate place to do this bit of clean up since it cleans up config on the guest